python/GafferDispatchUI/LocalJobs.py

```diff
@@ -0,0 +1,365 @@
+
+	def _children( self, canceller ) :
+
+		if self.isLeaf() :
+			return []
+
+		result = []
+		for index, job in enumerate( self.__jobPool.jobs() ) :
+			result.append(
+				_LocalJobsPath(
+					jobPool = self.__jobPool,
+					job = job,
+					path = [ str( index ) ],
+				)
+			)
+
+		return result
+
+class _StatusColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		status = path.property( "localDispatcher:status", canceller )
+		return GafferUI.PathColumn.CellData(
+			icon = f"localDispatcherStatus{status}.png",
+			toolTip = status,
+			sortValue = int( GafferDispatch.LocalDispatcher.Job.Status[status] )
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "Status" )
+
+class _RunningTimeColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		seconds = int( path.property( "localDispatcher:runningTime", canceller ) )
+
+		return GafferUI.PathColumn.CellData(
+			value = "{:02}:{:02}:{:02}".format(
+				seconds // 3600,
+				(seconds % 3600) // 60,
+				seconds % 60
+			),
+			sortValue = seconds
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "Running Time" )
+
+class _CPUUsageColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		cpu = path.property( "localDispatcher:cpuUsage", canceller )
+
+		return GafferUI.PathColumn.CellData(
+			value = f"{cpu:.2f}" if cpu is not None else "---",
+			sortValue = cpu if cpu is not None else 0.0,
+			toolTip = "CPU usage for current batch"
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "CPU" )
+
+class _MemoryUsageColumn( GafferUI.PathColumn ) :
+
+	def cellData( self, path, canceller ) :
+
+		memory = path.property( "localDispatcher:memoryUsage", canceller )
+
+		return GafferUI.PathColumn.CellData(
+			value = "{:.2f}GB".format( memory / (1024 ** 3) ) if memory is not None else "---",
+			sortValue = memory if memory is not None else 0,
+			toolTip = "Memory usage for current batch"
+		)
+
+	def headerData( self, canceller ) :
+
+		return GafferUI.PathColumn.CellData( value = "Memory" )
+
+class LocalJobs( GafferUI.Editor ) :
+
+	def __init__( self, scriptNode, **kw ) :
+
+		splitContainer = GafferUI.SplitContainer( borderWidth = 8 )
+		GafferUI.Editor.__init__( self, splitContainer, scriptNode, **kw )
+
+		jobPool = GafferDispatch.LocalDispatcher.defaultJobPool()
+
+		with splitContainer :
+
+			with GafferUI.ListContainer( spacing = 4 ) :
+
+				self.__jobListingWidget = GafferUI.PathListingWidget(
+					_LocalJobsPath( jobPool ),
+					columns = (
+						_StatusColumn(),
+						GafferUI.PathListingWidget.StandardColumn( "Name", "localDispatcher:jobName", sizeMode = GafferUI.PathColumn.SizeMode.Stretch ),
+						GafferUI.PathListingWidget.StandardColumn( "Id", "localDispatcher:id" ),
+						GafferUI.PathListingWidget.StandardColumn( "Start Time", "localDispatcher:startTime" ),
```

Comment:
The start time column is displaying oddly for me. It appears to be considering my local time as if it were UTC and then displays it with my local timezone offset from UTC applied, so the displayed start time is 8 hours earlier than my system clock. For example, if a job started at 9pm PST, the start time column will show 1pm. The start time listed in the Properties tab is correct, it's just the column display that is wrong.

I've tried this on Windows as well, with the same outcome.

If I change my system timezone to GMT/UTC then the start time column correctly matches my system clock which would explain why you wouldn't have run into this, though you should be able to reproduce it by shifting your own system time zone away from GMT...

Reply:
Following Great Britain's glorious exit from the European Union, Gaffer will be representing time solely as god intended, in GMT. I'm sure you will appreciate the simplification and reduction in red tape this brings.

Fixed in https://github.com/GafferHQ/gaffer/pull/5557/commits/62ab2167c8fb055fa40a9cf221feb1b49b2994a2 :)